libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1485,7 +1485,7 @@ Json::Value StandardCompiler::compileYul(InputsAndSettings _inputsAndSettings)
 						*o.bytecode,
 						o.sourceMappings.get(),
 						Json::arrayValue,
-						false,
+						objectKind == "deployedBytecode",
```

Comment:
We have a similar switch above. Can you create a local bool flag that does the string comparison only once, or iterate over `{true, false}`?